.clang-tidy

```diff
@@ -28,14 +31,17 @@ Checks:
     hicpp-signed-bitwise,
     misc-*,
     modernize-*,
+    -modernize-use-trailing-return-type,
     performance-*,
     portability-std-allocator-const,
     portability-simd-intrinsics,
     readability-*,
-    -readability-avoid-const-params-in-decls
+    -readability-avoid-const-params-in-decls,
+    -readability-function-cognitive-complexity
 WarningsAsErrors:
     boost-*,
     bugprone-*,
+    -bugprone-easily-swappable-parameters,
```

Comment:
Consider removing this from this section as it is good to think about when designing new APIs.

Reply:
Removed.